src/VCR/Storage/Yaml.php

```diff
@@ -58,8 +58,10 @@ public function storeRecording(array $recording)
     public function next()
     {
         $recording = $this->yamlParser->parse($this->readNextRecord());
-        $this->current = $recording[0];
-        ++$this->position;
+        if (!is_null($recording)) {
```

Comment:
It shouldn't happen, but if `$recording = 'some string'`, unintended things will happen.   I'd suggest:
```
if (is_array($recording) && count($recording))
```